pkg/k8sclient/resources/node/node.go

```diff
@@ -72,7 +72,8 @@ func (c *Client) cordonUnCordon(ctx context.Context, nodename string, cordon boo
 	if patchErr == nil {
 		patchOptions := metav1.PatchOptions{}
 		_, err = c.Interface.Patch(ctx, node.Name, types.StrategicMergePatchType, patchBytes, patchOptions)
-	} else {
+	}
+	if patchErr != nil || err != nil {
 		updateOptions := metav1.UpdateOptions{}
```

Comment:
Can we not move the Interface.Update right after the Interface.Patch if err != nil. Two if loops seem a bit redundant here. just a suggestion.

Reply:
It looks like @gilltaran made changes so that Update() is tried even when there is a failure from Patch().

Reply:
I see it, but err can only be set if returned by interface.Patch, checking for it right after makes more sense than another if block.

Reply:
Based on what I see, the second 'if' block is covering 2 scenarios. If we avoid that second block, there will be code duplication then.